low-level/files/opt/seagate/sspl/setup/sspl_test.py

```diff
@@ -76,85 +76,96 @@ def process(self):
+            shutil.copyfile(service_file_path_src,
+                            f'{service_file_path_des}/dummy_service.service')
+            cmd= "systemctl daemon-reload"
+            _, error, returncode = SimpleProcess(cmd).run()
+            if returncode !=0:
+                    print(f"failed to execute '{cmd}', systemctl will be unable"\
+                        f" to manage the dummy_service.service \nError: {error}")
+
+            self.dbus_service.enable(service_name)
+            self.dbus_service.start(service_name)
+
+            service_list = Conf.get(SSPL_CONFIG_INDEX, "SERVICEMONITOR>monitored_services")
+            service_list.append(service_name)
+            Conf.set(SSPL_CONFIG_INDEX, "SERVICEMONITOR>monitored_services",
+                service_list)
+
+            threshold_inactive_time_original = Conf.get(SSPL_CONFIG_INDEX,
+                                        "SERVICEMONITOR>threshold_inactive_time")
+            threshold_inactive_time_new = 30
+            Conf.set(SSPL_CONFIG_INDEX, "SERVICEMONITOR>threshold_inactive_time",
+                threshold_inactive_time_new)
+            Conf.save(SSPL_CONFIG_INDEX)
+
+            # TODO: Convert shell script to python
+            # from cortx.sspl.sspl_test.run_qa_test import RunQATest
+            # RunQATest(self.plan, self.avoid_rmq).run()
+            CMD = "%s/run_qa_test.sh %s %s" % (TEST_DIR, self.plan, self.avoid_rmq)
+            try:
+                output, error, rc = SimpleProcess(CMD).run(realtime_output=True)
+            except KeyboardInterrupt:
+                rc = 1
+                error = "KeyboardInterrupt occurred while executing sspl test."
+            # Restore the original path/file & service, then throw exception
+            # if execution is failed.
+            service_list.remove(service_name)
+            Conf.set(SSPL_CONFIG_INDEX,"SERVICEMONITOR>monitored_services",
+                service_list)
+            Conf.set(SSPL_CONFIG_INDEX, "SERVICEMONITOR>threshold_inactive_time",
+                threshold_inactive_time_original)
+            Conf.set(SSPL_CONFIG_INDEX, "SYSTEM_INFORMATION>global_config_copy_url",
+                sspl_global_config_url)
+            Conf.save(SSPL_CONFIG_INDEX)
+            shutil.copyfile(sspl_test_backup, sspl_test_file_path)
+            if rc != 0:
+                raise Exception("%s - ERROR: %s - CMD %s" % (self.name, error, CMD))
+        else:
+            # TODO: Convert shell script to python
+            # from cortx.sspl.sspl_test.run_qa_test import RunQATest
+            # RunQATest(self.plan, self.avoid_rmq).run()
+            try:
+                CMD = "%s/run_qa_test.sh %s %s" % (TEST_DIR, self.plan, self.avoid_rmq)
+                output, error, returncode = SimpleProcess(CMD).run(realtime_output=True)
+            except KeyboardInterrupt:
+                print("KeyboardInterrupt occurred while executing sspl test.")
```

Comment:
generic exception also in sspl_setup (or somewhere) to catch all exceptions, and print rc?

Reply:
Done. defined TestException class in same file, to catch all the exception.